src/scope.md

```diff
@@ -0,0 +1,132 @@
+- Semantic properties of the runtime behavior.
+- Built-in attributes.
+- Built-in types.
+- *Language items* from the standard library, which directly interface with the language (such as operators desugaring to trait calls).
+- All [editions] of Rust.
+
+## Out of scope
+
+The Reference does not cover:
+
+- Contents of the standard library, unless required to describe other language aspects.
+    - The standard library is described in [the standard library API documentation][std].
+- Introductions or guides to the language. Background familiarity with the language is assumed. A separate [book] is available to help acquire such background familiarity.
+- Recommendations on how to write Rust code.
+    - Exceptional cases may be included in a [note] or [warning].
+- Specifics of tooling provided by the Rust organization, such as `rustc` or Cargo.
+    - `rustc` has its own [book][rustc book]. Cargo has a [book][cargo book] that contains a [reference][cargo reference].
+    - In exceptional circumstances, notes may refer to these tools when they have significant relevance.
+- Specifics of which targets exist, or properties of specific targets. See [target-specific behavior] for more details.
+- Unstable features only available on the [nightly channel].
+    - For documentation of unstable features, see the [Unstable Book].
+    - Occasionally the Reference may refer to an unstable feature because it is important to understand the model of stable code.
+- Future changes, with the following exceptions:
+    - Language constructs that are designed to specifically allow for future extensions. For example, some syntax is reserved for future use, or some values such as config predicates may be documented as *open*, meaning they may be extended in the future.
+    - Notes for [incomplete or undecided behavior] may refer to the future intentions.
+    - Notes about deviation from [intended behavior] may mention that the behavior may change.
+- Version history or past behavior.
+    - Changes in Rust releases are recorded in the [Release Notes]. Prior releases of the Reference are available as described in [Rust releases].
+- Historical documentation.
+    - This includes documentation such as [RFCs], design documents, reports, blog posts, issues, pull requests, etc.
+- Rationale for most aspects of the language.
+    - Rationale may be included in a note in exceptional situations, particularly when a feature is not obvious or has especially useful relevance.
+- `rustc` lints.
+    - Lints are documented in the [rustc book][lints].
+    - Lints that have a specific relation to some part of the language may be referred to by name. For example, the `expect` attribute can issue the `unfulfilled_lint_expectations` lint. However, the exact behavior of when the lint is issued is usually not documented in the Reference.
+    - Notes may refer to lints when the lint has exceptional relevance to some aspect of the language.
```

Comment:
I'm wondering if we could provide further guidance here to mention [deny-by-default lints](https://doc.rust-lang.org/nightly/rustc/lints/listing/deny-by-default.html)? (The non-FCW ones at least.) Since they are relatively uncommon, and are deny because they are important and we want to steer people away from those hazards, maybe that is a good guideline to say that those are good examples of ones to mention in notes?

Unfortunately it gets a little difficult for some lints like `unconditional_panic` because that one lint covers a lot of different scenarios, and I don't think we want to document all of them (at least not here). (Because that lint is to some degree a best-effort thing, can change frequently over time, and I assume is pretty complicated.)